pkg/server/span_download.go

```diff
@@ -63,7 +66,19 @@ func (s *systemStatusServer) DownloadSpan(
 	remoteRequest := *req
 	remoteRequest.NodeID = "local"
 	nodeFn := func(ctx context.Context, status serverpb.RPCStatusClient, _ roachpb.NodeID) (*serverpb.DownloadSpanResponse, error) {
-		return status.DownloadSpan(ctx, &remoteRequest)
+		var nodeResp *serverpb.DownloadSpanResponse
+		var err error
+		for r := retry.StartWithCtx(ctx, retry.Options{
+			InitialBackoff: time.Millisecond * 100,
+			MaxBackoff:     time.Second,
+			MaxRetries:     perNodeMaxDownloadAttempts - 1,
+		}); r.Next(); {
+			nodeResp, err = status.DownloadSpan(ctx, &remoteRequest)
+			if err == nil && len(resp.Errors) == 0 {
+				break
+			}
+		}
```

Comment:
I sometimes am glad when I remember to log the err in the case where we are tying again since otherwise it can be a little opaque when we just see the rpcs are higher than we expect. /2¢

Reply:
Yeah that's a good point, I'll add that too